Travis failing to merge seems no problem - it seems git got confused over a trivial thing.

However - the implementation does not seem to work for me (Win 8.1). The toolbar is fixed to the three options `Up`, `New N`, `New F`, and whatever I switch tab to, stay there. I get repeatedly this error:

``` python
Traceback (most recent call last):
  File "noteorganiser/NoteOrganiser.py", line 271, in showActiveToolBar
    self.library.toolbar.setVisible(False)
AttributeError: 'Library' object has no attribute 'toolbar'
```

Surprisingly enough, the tests pass.

--- a/noteorganiser/NoteOrganiser.py
+++ b/noteorganiser/NoteOrganiser.py
@@ -261,9 +261,9 @@ class NoteOrganiser(QtGui.QMainWindow):
         activeTab = self.tabs.tabText(tabIndex)
         # activate, if there's a toolbar in library / editing
         if activeTab == "&Library":
-            self.library.toolbar.setVisible(True)
+            self.library.shelves.toolbar.setVisible(True)
         else:
-            self.library.toolbar.setVisible(False)
+            self.library.shelves.toolbar.setVisible(False)
         if activeTab == "&Editing":
             self.editing.toolbar.setVisible(True)
         else:

--- a/noteorganiser/frames.py
+++ b/noteorganiser/frames.py
@@ -790,6 +790,7 @@ class Shelves(CustomFrame):
     def updateUpAction(self):
         """
         update the state of the toolbar action 'Up'
+
         active if not in root
         """
         self.upAction.setDisabled(self.info.level == self.info.root)

--- a/noteorganiser/frames.py
+++ b/noteorganiser/frames.py
@@ -175,6 +175,21 @@ class Editing(CustomFrame):
             self.previewAction.triggered.connect(self.preview)
             self.toolbar.addAction(self.previewAction)
 
+            # separator between general and notebook specific actions
+            self.toolbar.addSeparator()
+
+            # save the Text in the current notebook editor
+            self.saveAction = QtGui.QAction(self)
+            self.saveAction.setIconText('&Save')
+            self.saveAction.triggered.connect(self.saveText)
+            self.toolbar.addAction(self.saveAction)
+
+            # reload the Text in the current notebook editor
+            self.readAction = QtGui.QAction(self)
+            self.readAction.setIconText('&Reload')
+            self.readAction.triggered.connect(self.loadText)
+            self.toolbar.addAction(self.readAction)
+
         # Global horizontal layout
         hbox = QtGui.QHBoxLayout()
 
@@ -283,6 +298,16 @@ class Editing(CustomFrame):
         # recover the current editor
         editor = self.tabs.currentWidget()
         editor.resetSize()
+
+    def loadText(self):
+        """ reload the text in the current notebook """
+        notebook = self.tabs.currentWidget()
+        notebook.loadText()
+
+    def saveText(self):
+        """ save the text in the current notebook """
+        notebook = self.tabs.currentWidget()
+        notebook.saveText()
 
 
 class Preview(CustomFrame):
@@ -802,21 +827,6 @@ class TextEditor(CustomFrame):
 
     def initUI(self):
         """top menu bar and the text area"""
-        # Menu bar
-        menuBar = QtGui.QHBoxLayout()
-
-        self.saveButton = QtGui.QPushButton("&Save", self)
-        self.saveButton.clicked.connect(self.saveText)
-
-        self.readButton = QtGui.QPushButton("&Reload", self)
-        self.readButton.clicked.connect(self.loadText)
-
-        menuBar.addWidget(self.saveButton)
-        menuBar.addWidget(self.readButton)
-        menuBar.addStretch(1)
-
-        self.layout().addLayout(menuBar)
-
         # Text
         self.text = CustomTextEdit(self)
         self.text.setTabChangesFocus(True)

